activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -166,7 +166,14 @@ def begin_transaction(options = {})
       end
 
       def commit_transaction
-        @stack.pop.commit
+        transaction = @stack.pop
```

Comment:
I think this would be several times less confusing if it was named `inner_transaction` or similar; having `transaction` and `current_transaction` be different things is nigh unreadable to me.